buildSrc/src/main/kotlin/codegen/math/GenerateMath.kt

```diff
@@ -0,0 +1,249 @@
+package codegen.math
+
+import com.squareup.kotlinpoet.*
+import com.squareup.kotlinpoet.KModifier.*
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+
+@Suppress("UnstableApiUsage")
+open class GenerateMath : DefaultTask() {
+	@OutputDirectory
+	val outputDir = project.objects.directoryProperty()
+
+	@get:OutputDirectory
+	val commonDir = outputDir.file("common")
+
+
+	private val packageName = "com.kgl.math"
+
+	private val typeList = listOf(
+		BYTE to "0",
+		SHORT to "0",
+		INT to "0",
+		LONG to "0L",
+		U_BYTE to "0U",
+		U_SHORT to "0U",
+		U_INT to "0U",
+		U_LONG to "0UL",
+		FLOAT to "0f",
+		DOUBLE to "0.0"
+	)
+
+
+	@TaskAction
+	fun generate() {
+		vectorTypes()
+	}
+
+	private val allComponents = listOf(1 to "x", 2 to "y", 3 to "z", 4 to "w")
+	private val arithmetics = listOf("plus" to "+", "minus" to "-", "times" to "*", "div" to "/")
+
+	private fun vectorTypes() {
+		for (componentCount in 2..4) {
+			val components = allComponents.take(componentCount)
+
+			buildFile(packageName, "Vector$componentCount") {
```

Comment:
Can we put each vector sealed hierarchy in separate files? It would make it a bit easier to navigate.

Reply:
In separate Generated files, or in separate codegen source files? The first is already done.

Reply:
Separate the generated files. Even more then it is now.

Reply:
Oh, as in move `FloatVector2` and `DoubleVector2` to separate files? Yeah, no problem. 👌

Reply:
https://github.com/Dominaezzz/kgl/pull/19/commits/f7b4dfa6f636fd8074adcaaa8e246e4f5d5e12a4